src/main/scala/io/iohk/ethereum/network/p2p/messages/PV63.scala

```diff
@@ -75,13 +75,17 @@ object PV63 {
 
       override def decode(rlp: RLPEncodeable): MptNode = rlp match {
         case rlpList: RLPList if rlpList.items.length == BranchNodeChildLength + 1 =>
-          MptBranch(rlpList.items.take(BranchNodeChildLength).map { bytes =>
-            val v = rlpDecode[ByteString](bytes)
-            if (v.nonEmpty && v.length < 32) {
-              Right(MptValue(v))
-            } else {
-              Left(MptHash(v))
-            }
+          MptBranch(rlpList.items.take(BranchNodeChildLength).map {
+            case bytes: RLPValue =>
+              val v = rlpDecode[ByteString](bytes)
+              if (v.nonEmpty && v.length < 32) {
+                Right(MptValue(v))
+              } else {
+                Left(MptHash(v))
+              }
+
+            case value:RLPEncodeable =>
```

Comment:
Shouldn't this be `case value: RLPList`? Is there any case when the `RLPEncodeable` matching is necessary?